src/storage/lock_manager/util.rs

```diff
@@ -2,32 +2,45 @@
 
 use super::Lock;
 use crate::storage::mvcc::Error as MvccError;
-use crate::storage::txn::Error as TxnError;
+use crate::storage::txn::{Error as TxnError, ProcessResult};
 use crate::storage::Error as StorageError;
 use crate::storage::Key;
-use std::collections::hash_map::DefaultHasher;
-use std::hash::{Hash, Hasher};
+use farmhash;
 
 pub fn extract_lock_from_result(res: &Result<(), StorageError>) -> Lock {
     match res {
         Err(StorageError::Txn(TxnError::Mvcc(MvccError::KeyIsLocked { key, ts, .. }))) => Lock {
             ts: *ts,
             hash: gen_key_hash(&Key::from_raw(&key)),
         },
-        _ => panic!("unsupported mvcc error"),
+        _ => panic!("unexpected mvcc error"),
     }
 }
 
+// TiDB uses the same hash algorithm.
 pub fn gen_key_hash(key: &Key) -> u64 {
-    let mut s = DefaultHasher::new();
-    key.hash(&mut s);
-    s.finish()
+    farmhash::fingerprint64(&key.to_raw().unwrap())
```

Comment:
Please give some benchmark for farmhash.

Reply:
```
running 2 tests
test bench_farmhash ... bench:          19 ns/iter (+/- 0)
test bench_std_hash ... bench:          35 ns/iter (+/- 1)
```

TiDB will check `deadlock_key_hash` to judge is it a single statement deadlock. So TiKV should use the same hash algorithm with TiDB.